### ISSUE TITLE
`import/export` failed to find named exports from indirect jsx (or tsx) files

### ISSUE THREAD

**Issue by YF1999** [benmosher/eslint-plugin-import#2093]
There are three files as following:

```ts
// module/feature.tsx or module/feature.jsx
export function func() {
    console.log('Hello world');
}
```

```ts
// module/index.ts
export * from './feature';
```

```ts
// index.ts
export * from './module';
```

Then an error `No named exports found in module './module'. eslint(import/export)` will be raised in the outer `index.ts`.

**Comment by ljharb** [comment]
Do you have `.ts` and `.jsx` both in the `import/extensions` setting?

This test cases passes, so I'll close the issue when I push up the commit that adds it.
